### ISSUE TITLE
Fullscreen switch via menu option not working

### ISSUE THREAD

**Issue by Morlic-fo** [freeorion/freeorion#964]
### Environment
- **FreeOrion Version**: 2016-09-09-3058052 (RC1 0.4.6)
- **Operating System**: Win7
- **Graphic card used**: GT520M
- **Fetched as**  \* Weekly development build
### Description

Clicking the ingame option menu point "Full Screen" does not have effect. Neither does the game change the setting immediately nor after a restart.

Switching using Alt+Enter works for me as well as starting the game with  --fullscreen argument.

I deleted Config.xml before the test and have no persistent_config.xml
### Expected Result

a) Game switches between Full Screen / Windowed mode immediately
b) Game switches between Full Screen / Windowed mode on restart
### Steps to reproduce
- Open Game in windowed mode
- Open ingame options menu, activate Video > Fullscreen
- Game does not change to fullscreen.
- Restart game using the .exe or a shortcut
- Game opens in windowed mode again

**Comment by geoffthemedio** [comment]
Did you click "Apply" after changing the check box on the options screen video tab?

**Comment by Morlic-fo** [comment]
No. I clicked "Done". Clicking "Apply" works of course.

I am the dumbest assumable user. Windows (and most other programs I know) tend to automatically "Apply" all changes when pressing "Ok" instead of "Cancel" or similar. Considering that the changed selection is stored when closing the menu and opening  it again, I expect it to at least apply on restart.

If it is indeed intended to only work if "Apply" is pressed explicitly, the tooltip should be updated accordingly (and the window default size changed I suppose). From my point of view, though, it would be more intuitive if pressing "Done" implies an "Apply".
Also note that playing on resolution 1366x768, the "Apply" button is by default visible only after rescaling the option window or scrolling down.

**Comment by Vezzra** [comment]
> I am the dumbest assumable user.

Well, welcome to the club, if that makes you feel any better. 😉 I occasionally walked into that trap too - made some adjustments to the screen settings, hit "Done" and then wondered why nothing changed, went back, repeated my actions, same result and thought "Oh crap, something's broken."

Until I realized that there is this "Apply" button I didn't care to pay attention to, and felt really, really stupid... 😳 

> it would be more intuitive if pressing "Done" implies an "Apply"

In that case we need to provide a "Cancel" botton too, because otherwise you wouldn't be able _not_ to apply any changes you made to the settings and want to back out of it.

**Comment by Vezzra** [comment]
Anyway, changing the behaviour of the buttons in the options dialog should probably be a separate issue, so can we close this as "wontfix"/"invalid" (whatever you think fits better)?

**Comment by adrianbroher** [comment]
> In that case we need to provide a "Cancel" botton too, because otherwise you wouldn't be able not to apply any changes you made to the settings and want to back out of it.

I would say the 'Apply' button is placed wrong. It should be placed in the bottom section of the window, not within the 'Video' tab. It's the only intro menu window where we made this odd layout.

> Anyway, changing the behaviour of the buttons in the options dialog should probably be a separate issue, so can we close this as "wontfix"/"invalid" (whatever you think fits better)?

No, I would keep it open even if this isn't a "does not work" bug but a UI flaw.

**Comment by Vezzra** [comment]
> No, I would keep it open even if this isn't a "does not work" bug but a UI flaw.

Well, it's just that the issue title and OP text don't really describe the actual problem correctly. If we keep this issue, we should at least adjust the issue title to something more fitting?

**Comment by Vezzra** [comment]
> Well, it's just that the issue title and OP text don't really describe the actual problem correctly. If we keep this issue, we should at least adjust the issue title to something more fitting?

Bump?
